lib/ex_unit/lib/ex_unit/on_exit_handler.ex

```diff
@@ -18,8 +18,9 @@ defmodule ExUnit.OnExitHandler do
     :ets.insert(@name, {pid, nil, []})
   end
 
-  @spec add(pid, term, (() -> term)) :: :ok | :error
-  def add(pid, name_or_ref, callback) when is_pid(pid) and is_function(callback, 0) do
+  @spec add(pid, term, (() -> term) | ((ExUnit.Test.t | ExUnit.TestModule.t) -> term)) ::
+    :ok | :error
+  def add(pid, name_or_ref, callback) when is_pid(pid) and (is_function(callback, 0) or is_function(callback, 1)) do
```

Comment:
We can probably remove the `is_function` checks from here since we already validate in the boundary.